proposals/004_Overlays.md

```diff
@@ -0,0 +1,178 @@
+
+```yaml
+overlay: 1.0.0
+info:
+  title: Structured Overlay
+  version: 1.0.0
+updates:
+- target: "@"
+    value:
+      info:
+        x-overlay-applied: structured-overlay
+      paths:
+        "/":
+          summary: "The root resource"
+          get:
+            summary: "Retrieve the root resource"
+            x-rate-limit: 100
+        "/pets":
+          get:
+            summary: "Retrieve a list of pets"
+            x-rate-limit: 100
+      components:
+      tags:
+```
+
+##### Targeted Overlays
+
+Alternatively, where only a small number of updates need to be applied to a large document, each [Update Object](#updateObject) can be more targeted.
+
+```yaml
+overlay: 1.0.0
+info:
+  title: Structured Overlay
+  version: 1.0.0
+updates:
+- target: paths."/foo".get
+    value:
+        description: This is the new description
+- target: paths."/bar".get
+    value:
+        description: This is the updated description
+- target: paths."/bar"
+    value:
+        post:
+            description: This is an updated description of a child object
+            x-safe: false
+```
+
+##### Wildcard Overlays Examples
+
+One significant advantage of using the JMESPath syntax that it allows referencing multiple nodes in the target document.  This would allow a single update object to be applied to multiple target objects using wildcards.
+
+```yaml
+overlay: 1.0.0
+info:
+  title: Update many objects at once
+  version: 1.0.0
+updates:
+- target: paths.*.get
+    value:
+      x-safe: true
+- target: paths.*.get.parameters[?name=='filter' && in=='query']
+    value:
+      schema:
+        $ref: "/components/schemas/filterSchema"
+```
+
+##### Array Modification Examples
+
+Due to the fact that we can now reference specific elements of the parameter array, it does open the possibilty to being able to add parameters and potentially remove them using a `null` value.
+
+```yaml
+overlay: 1.0.0
+info:
+  title: Add an array element
+  version: 1.0.0
+updates:
+- target: paths.*.get.parameters[length(@)]
+    value: 
+      name: newParam
+      in: query
+```
+
+```yaml
+overlay: 1.0.0
+info:
+  title: Remove a array element
+  version: 1.0.0
+updates:
+- target: $.paths[*].get.parameters[? name == 'dummy']
+    remove: true
+```
+
+
+## Proposal Summary
+
+### Benefits
+
+- This approach addresses the two distinct approaches of structured overlay vs targeted overlay which suits distinct but equally valid scenarios.
+- Addresses the problem of modifying the parameters array and removes the need to replace the entire array when a small change is required.
+- Allows sets of related overlays to be stored in a same file.
+- Enables updating a set of objects based on a pattern. This might be an effective way of apply common behaviour across many operations in an API.
+
+### Challenges
+- Tooling will need a JMESPath implementation.
```

Comment:
Suggest adding a "Alternatives considered" paragraph as per the template, which explains why JSONPath was considered and JMESPath preferred (no formal published specification, and security / compatibility concerns over evaluation mechanism).